src/Base/UnitsApi.cpp

```diff
@@ -139,23 +127,34 @@ void UnitsApi::setSchema(UnitSystem system)
                                        // Quantity (e.g. mi=1.8km rather then 1.6km).
 }
 
-QString UnitsApi::toString(const Base::Quantity& quantity, const QuantityFormat& format)
+std::string UnitsApi::toString(const Base::Quantity& quantity, const QuantityFormat& format)
 {
-    QString value = QString::fromLatin1("'%1 %2'")
-                        .arg(quantity.getValue(), 0, format.toFormat(), format.precision)
-                        .arg(quantity.getUnit().getString());
-    return value;
+    return '\'' + toNumber(quantity, format) + ' ' + quantity.getUnit().getString() + '\'';

```

Comment:
Creating a std::string this way is very in-efficient. It creates five tmp. string objects. You should use one instance and add the further strings with append()

Reply:
Replaced with `stringstream`.